Content.Client/UserInterface/Systems/Ghost/Controls/Roles/GhostRolesEui.cs

```diff
@@ -77,35 +76,45 @@ public override void HandleState(EuiStateBase state)
 
             if (state is not GhostRolesEuiState ghostState)
                 return;
+
+            // We must save BodyVisible state, so all Collapsible boxes will not close
+            // on adding new ghost role.
+            // Save the current state of each Collapsible box being visible or not
+            var collapsibleStates = _window.SaveCollapsibleBoxesStates();
+
+            // Clearing the container before adding new roles
             _window.ClearEntries();
 
             var entityManager = IoCManager.Resolve<IEntityManager>();
             var sysManager = entityManager.EntitySysManager;
             var spriteSystem = sysManager.GetEntitySystem<SpriteSystem>();
             var requirementsManager = IoCManager.Resolve<JobRequirementsManager>();
 
+            // Grouping roles
             var groupedRoles = ghostState.GhostRoles.GroupBy(
                 role => (role.Name, role.Description, role.Requirements));
```

Comment:
Oh hey so this isn't your fault and I'm not gonna demand you fix it, but I just wanna point out this `role.Requirements` value doesn't work at all as an equality key and this code is broken. Lovely.

Reply:
I will try to fix it

Reply:
I left a TODO for it in Eui